app/runtime/pom.xml

```diff
@@ -91,6 +91,12 @@
   <dependencyManagement>
     <dependencies>
       <!-- Syndesis Runtime Modules -->
+      <dependency>
+        <groupId>io.syndesis</groupId>
+        <artifactId>core</artifactId>
```

Comment:
Could we avoid this dependency ? Seems to be a bit much a dependency only for the KeyGenerator. Would be ok for me even to copy the class instead (doesn't seem to be changed often).

Reply:
What's in core that seems so risky to share?

Reply:
Not really risky, but a question how to interprete "common": Common to the backend infra (rest, verifier, ...) or common to the integration runtime, or both ? Architectural-wise I really would like to keep that separated so that it can be evolve differently. Remember the runtime and rest will always share the same latest version number, so when you change KeyGenerator because you need it differently in the backend you have to change the integration runtime, too (and have to test whether it works for both use cases). Also, it would then require that your infra changes potentially lead to an integration upgrade maybe only because of this change (which would not be needed otherwise), if one want to keep the runtime on the latest versions (which you want, because you do not want to track different versions for your runtime).

At the end, its the same argument as they have for Microservices. The less dependencies the better. My POV is, that dependencies are inevitable if required. But they are always heavyweight (see the use case above), so we should not introduce dependencies lightly. Especially not for accessing a single small sized class.

Reply:
I don't mind making copies of stuff it it will make it easier to reason about, but this does not seem like we are running into these issues yet.  Could we fork the copy later if we need to?

Reply:
ok